src/psyclone/psyGen.py

```diff
@@ -1579,17 +1556,24 @@ def _insert_module_inlined_kernel(self, f2pygen_parent=None):
                     module = module.parent
                 search = PSyIRGen(module, self.get_kernel_schedule()).root
                 for child in module.children:
-                    if isinstance(child, PSyIRGen):
-                        if child.root == search:
-                            # If there is an exact match (the implementation is
-                            # the same), it is safe to continue.
-                            break
+                    if isinstance(child, PSyIRGen) \
+                            and child.root.name == search.name:
+                        if child.root != search:
+                            raise NotImplementedError(
+                                f"Can not inline subroutine '{self._name}' "
+                                f"because another, different, subroutine with "
+                                f"the same name already exists and versioning "
+                                f"of module-inlined subroutines is not "
+                                f"implemented yet.")
+                        else:
+                            # Found and matches expected routine
+                            return
                 else:
-                    raise NotImplementedError(
-                        "Can not inline subroutine '{0}' because another, "
-                        "different, subroutine with the same name already "
-                        "exists and versioning of module-inlined subroutines "
-                        "is not implemented yet.".format(self._name))
+                    # Not found, but the symbol already existed, this means it
+                    # is a second psy.gen call with starts the module_gen from
```

Comment:
This sentence doesn't make sense. In particular the word "with". Should it be "which"?

Reply:
Updated the whole sentence